spec/prometheus/client/data_stores/direct_file_store_spec.rb

```diff
@@ -267,6 +292,57 @@
     end
   end
 
+  context "with a metric that takes MOST_RECENT instead of SUM" do
+    it "reports the most recently written value from different processes" do
+      metric_store1 = subject.for_metric(
+        :metric_name,
+        metric_type: :gauge,
+        metric_settings: { aggregation: :most_recent }
+      )
+      metric_store2 = subject.for_metric(
+        :metric_name,
+        metric_type: :gauge,
+        metric_settings: { aggregation: :most_recent }
+      )
+
+      allow(Process).to receive(:pid).and_return(12345)
+      metric_store1.set(labels: { foo: "bar" }, val: 1)
+      allow(Process).to receive(:pid).and_return(23456)
+      metric_store2.set(labels: { foo: "bar" }, val: 3)
+      allow(Process).to receive(:pid).and_return(12345)
+      metric_store1.set(labels: { foo: "baz" }, val: 7)
+      allow(Process).to receive(:pid).and_return(23456)
+      metric_store2.set(labels: { foo: "baz" }, val: 2)
+      allow(Process).to receive(:pid).and_return(12345)
+      metric_store1.set(labels: { foo: "baz" }, val: 4)
+      allow(Process).to receive(:pid).and_return(23456)
```

Comment:
Not sure what these last two examples are here to test. I think all the behaviour is covered by the ones above?

Reply:
maybe i should leave only one of those two. What this is testing is a labelset that only exists in one of the stores, all the others are in both.